### DISCUSSION TITLE
Enrich the attributes of projection geometry

### DISCUSSION THREAD

**Discussion by LANGZHIZHEN** [astra-toolbox/astra-toolbox#529] [Q&A]
When I constructed the projection geometry using `proj_geom = astra.create_proj_geom('cone_vec', height, width, vectors)`, I found that only one projection Angle was involved, and the x,y, and z angles were not involved. When I reconstruct, the geometric information is given through the calibrated projection matrix. Usually, the ray source has a certain Angle at x,y, and z, which leads to a certain deviation in the reconstruction results.

**Comment by askorikov** [comment]
Hi! I don't think I get what you mean completely, but 'cone_vec' geometry is not defined by projection angles directly. Instead, it's parametrized by the source and detector center position, and rotation of the the detector plane as explained [here](https://astra-toolbox.com/docs/geom3d.html#projection-geometries). You need to calculate these parameters from your angle description.

**Reply by LANGZHIZHEN** [reply]
What I mean is that my ray source is at an Angle to the detector in the x,y, and z directions. I see that the code only provides projAngle, maybe my ray source is not orthogonal to the center of rotation.
